sklearn/feature_selection/_univariate_selection.py

```diff
@@ -297,15 +297,30 @@ def f_regression(X, y, *, center=True):
     else:
         X_norms = row_norms(X.T)
 
+    if np.any(X_norms == 0) or np.std(y) == 0:
```

Comment:
`np.ptp` may be better for checking for a constant target.

Another note, having feature selection error for cross validation when one of the folds have a constant target feels strange?

Reply:
Thanks @thomasjpfan, I did not know the `np.ptp` function!

Regarding this error in a feature selection process for cross validation, I think that is a kind of weird if the user provides a non-constant target but, during the splitting, one of the folds contains a constant target. The same applies for attributes.

WDYT?

Reply:
If the regression target was discrete, I can see this happening randomly. (Discrete such as age by every 10 years: 10, 20, 30, 40, etc)

I wouldn't want feature selection to error in these cases. I am wondering if there is a reasonable thing for `SelectKBest` to do with `score_func` raises an error.

Reply:
The main issue that I can see here is: how can we select the k-best features in `SelectKBest` without actually obtaining any scores from the `score_func` function (maybe to assign a zero-score and let `SelectKBest` based on index?

Reply:
Another possible solution could be to raise a `(User)Warning` instead of a `ValueError` and return `np.nan`. Then, `SelectKBest` will select based on index (due to the behaviour of `numpy.argsort` in these cases).

Reply:
I am okay with using `nan`. It is the current behavior:

```py
from sklearn.feature_selection import f_regression, SelectKBest
from sklearn.datasets import make_regression
import numpy as np

X, y = make_regression(n_features=4, random_state=42)
y = np.ones_like(y)
skb = SelectKBest(f_regression, k=2).fit(X, y)

skb.pvalues_
# array([nan, nan, nan, nan])

skb.scores_
# array([nan, nan, nan, nan])
```

Reply:
I will be happy to maintain the current behaviour.

Reply:
What do you think about removing both `ValueError` and handling the division by zero errors with `np.errstate`?

Reply:
I am okay with `errstate` to hide the warnings. We should make sure that the behavior is the same as before:

```py
from sklearn.feature_selection import f_regression, SelectKBest
import numpy as np

X = np.asarray([[2,1], [2,0], [2, 10], [2, 4]])
y = np.asarray([0, 1, 1, 0])
f_regression(X, y)

(array([      nan, 0.2293578]), array([       nan, 0.67924985]))
```

I do not know if this is statistically correct, but it kind of makes sense to have the `nans` by 0 and the p-value be 0. In these cases, a constant features would provide no value.

Reply:
I would be happy to see `np.nan` by `0` too, although I do not know if this is statistically correct neither.